### PR TITLE
Added test cases for some models with 100% coverage

### PR DESCRIPTION
**Pull request by ahmadhasankhan**
```
app/models/word.rb | 100.0 % | 57 | 19 | 19 | 0 | 1.7
app/models/word_corpus.rb | 100.0 % | 17 | 2 | 2 | 0 | 1.0
app/models/word_lemma.rb | 100.0 % | 15 | 3 | 3 | 0 | 1.0
app/models/word_root.rb | 100.0 % | 15 | 3 | 3 | 0 | 1.0
app/models/word_stem.rb | 100.0 % | 15 | 3 | 3 | 0 | 1.0
spec/models/word_corpus_spec.rb | 100.0 % | 24 | 6 | 6 | 0 | 1.0
spec/models/word_lemma_spec.rb | 100.0 % | 26 | 9 | 9 | 0 | 1.0
spec/models/word_root_spec.rb | 100.0 % | 26 | 9 | 9 | 0 | 1.0
spec/models/word_spec.rb | 100.0 % | 142 | 65 | 65 | 0 | 1.4
spec/models/word_stem_spec.rb | 100.0 % | 26 | 9 | 9 | 0 | 1.0

```

### CODE REVIEW

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32931

File: Gemfile [hunk 1]
```diff
@@ -58,11 +58,13 @@ gem 'sitemap_generator'
...
   gem 'pry-byebug'
   gem 'pry-rails'
-  gem 'rspec-rails' # http://betterspecs.org/
+  gem 'rspec-rails', '~> 3.5'
+  gem 'factory_girl_rails', '~> 4.8', require: false
   gem 'guard-rspec', require: false
   gem 'parallel_tests'
   gem 'zeus'
   gem 'simplecov', :require => false
+  gem 'database_cleaner'
```
Comment (naveed-ahmad) [inline Gemfile]: Our DB is static, wonder why we ever wanted to clear it :) Even for the test environment. Will review the PR tomorrow inshAllah

**Comment by korabh** [comment]
Hey @ahmadhasankhan, I suggest to use [shoulda-matchers](https://github.com/thoughtbot/shoulda-matchers) instead of macro symbols.

**Comment by mmahalwy** [comment]
This is awesome! let us know when it's ready for review!

**Comment by ahmadhasankhan** [comment]
> Hey @ahmadhasankhan, I suggest to use shoulda-matchers instead of macro symbols.

Hey @korabh,
Sometimes back i went through a guide called [betterspecs](http://www.betterspecs.org), and I really liked that, so following the same.
Please have a look and you guys can suggest if something is not fit/suitable for this app.
http://www.betterspecs.org/#expect

File: Gemfile (hunk 1 above)
Comment (ahmadhasankhan) [reply Gemfile]: Pardon me if i sound stupid, actually i haven't gone through the whole codebase yet, I just started with Models.

When you say 
> Our DB is static

Does this mean we don't have any create operation ?

Comment (mmahalwy) [reply Gemfile]: I think what he means is we can run tests against the static db data. Quran data doesn't change and thus you don't need a db cleaner.

**Comment by naveed-ahmad** [comment]
@ahmadhasankhan Sorry for the late response, and thanks for PR. I would suggest following changes:
- Remove `FactoryGirl` and `DatabaseCleaner`
    --> DB is static, we can't and should never add/update data in any table using api.
- Use Shoulda for testing validation and associations 
   --> I'm fine using reflections for testing the associations, but love shoulda for its readability. How about following this template:

```ruby
RSpec.describe Verse, type: :model do
  context '#columns' do
    [ :chapter_id, :verse_number, :juz_number, ... ].each do |attr| 
      it { should have_db_column(attr) } 
    end
  end

  context '#associations' do
     it { should belong_to(:chapter) }
     it { should has_many(:words) }
     ...
  end
  
  context '#something else' do 
  end
end
```
- SQL query specs are not part of this PR but ill love to start testing queries count( this might help us improve performance ) https://github.com/brigade/db-query-matchers

File: Gemfile (hunk 1 above)
Comment (ahmadhasankhan) [reply Gemfile]: @mmahalwy Gotcha.
BTW where can i fid the DB dump ?

Thanks

Comment (naveed-ahmad) [reply Gemfile]: @ahmadhasankhan ping me on slack for dump

**Pull request closed without merging**